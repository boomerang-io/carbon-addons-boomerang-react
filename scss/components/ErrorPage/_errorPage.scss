@use '@carbon/themes/scss/tokens' as theme;
@use "../../styles/config" as *;
@use "../../styles/utils";

  .#{$prefix}--bmrg-error-page {
    align-items: center;
    color: theme.$text-secondary;
    display: flex;
    font-family: 'IBM Plex Sans', 'Helvetica Neue', Arial, Helvetica, sans-serif;
    flex-direction: column;
    margin: utils.rem(80px) auto;
    padding: 0 2rem;
    width: 100%;
  }

  .#{$prefix}--bmrg-error-page__header {
    color: theme.$support-error-inverse;
    font-size: utils.rem(32px);
    font-weight: 600;
    line-height: 42px;
    margin-bottom: utils.rem(30px);
    text-align: center;
  }

  .#{$prefix}--bmrg-error-page__title {
    color: inherit;
    font-size: utils.rem(24px);
    font-weight: 600;
    line-height: utils.rem(32px);
    text-align: center;
  }

  .#{$prefix}--bmrg-error-page__message {
    color: inherit;
    font-size: utils.rem(16px);
    line-height: utils.rem(20px);
    margin-top: utils.rem(12px);
    max-width: 60ch;
    text-align: center;

    a {
      color: theme.$background-brand;
      font-weight: 600;
      text-decoration: none;
    }
  }

  .#{$prefix}--bmrg-error-page__graphic {
    display: flex;
    justify-content: center;
    margin-top: utils.rem(40px);
    max-width: utils.rem(576px);
    width: 100%;

    svg {
      height: 100%;
      width: 100%;
    }
  }