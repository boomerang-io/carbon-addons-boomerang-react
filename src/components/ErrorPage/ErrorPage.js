import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { settings } from 'carbon-components';

const { prefix } = settings;

ErrorPage.propTypes = {
  className: PropTypes.string,
  graphic: PropTypes.node,
  header: PropTypes.node,
  message: PropTypes.node,
  style: PropTypes.object,
  title: PropTypes.node,
};

export default function ErrorPage({ className, graphic, header, message, style, title }) {
  return (
    <div className={cx(`${prefix}--bmrg-error-page`, className)} style={style}>
      {header && <header className={`${prefix}--bmrg-error-page__header`}>{header}</header>}
      {title && <h1 className={`${prefix}--bmrg-error-page__title`}>{title}</h1>}
      {message && <section className={`${prefix}--bmrg-error-page__message`}>{message}</section>}
      {graphic && <section className={`${prefix}--bmrg-error-page__graphic`}>{graphic}</section>}
    </div>
  );
}
