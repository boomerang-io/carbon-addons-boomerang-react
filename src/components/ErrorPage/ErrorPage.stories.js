import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';

import GraphicWrangler from '../GraphicWrangler';

import ErrorPageComponent from './index';

storiesOf('ErrorPage', module)
  .add('Default', () => {
    return (
      <ErrorPageComponent
        header={text('header', 'Header')}
        title={text('title', 'Title')}
        message={text('message', 'Message')}
        graphic={<GraphicWrangler />}
      />
    );
  })
  .add('message link', () => {
    return (
      <ErrorPageComponent
        title={text('title', 'Title')}
        message={
          <p>
            Hello there, <a href="https://useboomerang.io">use Boomerang!</a>
          </p>
        }
        graphic={<GraphicWrangler />}
      />
    );
  });
