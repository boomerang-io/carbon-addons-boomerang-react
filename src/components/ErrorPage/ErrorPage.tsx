/*
IBM Confidential
694970X, 69497O0
Â© Copyright IBM Corp. 2022, 2024
*/


import React from "react";
import cx from "classnames";
import { prefix } from "../../internal/settings";

export type Props = {
  className?: string;
  graphic?: React.ReactNode;
  header?: React.ReactNode;
  message?: React.ReactNode;
  style?: React.CSSProperties;
  title?: React.ReactNode;
};

export default function ErrorPage({ className, graphic, header, message, style, title }: Props) {
  return (
    <div className={cx(`${prefix}--bmrg-error-page`, className)} style={style}>
      {header && <header className={`${prefix}--bmrg-error-page__header`}>{header}</header>}
      {title && <h1 className={`${prefix}--bmrg-error-page__title`}>{title}</h1>}
      {message && <section className={`${prefix}--bmrg-error-page__message`}>{message}</section>}
      {graphic && <section className={`${prefix}--bmrg-error-page__graphic`}>{graphic}</section>}
    </div>
  );
}
