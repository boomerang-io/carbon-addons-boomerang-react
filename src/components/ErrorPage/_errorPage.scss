@include exports('addons-bmrg-errorPage') {
  .#{$prefix}--bmrg-error-page {
    align-items: center;
    color: $text-02;
    display: flex;
    font-family: 'IBM Plex Sans', 'Helvetica Neue', Arial, Helvetica, sans-serif;
    flex-direction: column;
    margin: rem(80px) auto;
    padding: 0 2rem;
    width: 100%;
  }

  .#{$prefix}--bmrg-error-page__header {
    color: $inverse-support-01;
    font-size: rem(32px);
    font-weight: 600;
    line-height: 42px;
    margin-bottom: rem(30px);
    text-align: center;
  }

  .#{$prefix}--bmrg-error-page__title {
    color: inherit;
    font-size: rem(24px);
    font-weight: 600;
    line-height: rem(32px);
    text-align: center;
  }

  .#{$prefix}--bmrg-error-page__message {
    color: inherit;
    font-size: rem(16px);
    line-height: rem(20px);
    margin-top: rem(12px);
    max-width: 60ch;
    text-align: center;

    a {
      color: $interactive-01;
      font-weight: 600;
      text-decoration: none;
    }
  }

  .#{$prefix}--bmrg-error-page__graphic {
    display: flex;
    justify-content: center;
    margin-top: rem(40px);
    max-width: rem(576px);
    width: 100%;

    svg {
      height: 100%;
      width: 100%;
    }
  }
}
