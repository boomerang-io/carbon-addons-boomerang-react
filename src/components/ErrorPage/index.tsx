export { default } from "./ErrorPage";
export type { Props } from "./ErrorPage";
