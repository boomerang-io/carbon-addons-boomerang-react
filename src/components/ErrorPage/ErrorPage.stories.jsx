import { text } from "@storybook/addon-knobs";
import GraphicWrangler from "../GraphicWrangler";
import ErrorPage from "./ErrorPage";

export default {
  title: "Errors/ErrorPage",
  component: ErrorPage,
  parameters: {
    docs: {
      description: {
        component:
          "Customizable full page error for Boomerang theme. Used as base component for core 403 and 404 error components",
      },
    },
  },
};

export const Default = (args) => {
  return (
    <ErrorPage
      header={text("header", "Header")}
      title={text("title", "Title")}
      message={text("message", "Message")}
      graphic={<GraphicWrangler />}
      {...args}
    />
  );
};

export const MessageLink = (args) => {
  return (
    <ErrorPage
      title={text("title", "Title")}
      message={
        <p>
          Hello there, <a href="https://useboomerang.io">use Boomerang!</a>
        </p>
      }
      graphic={<GraphicWrangler />}
      {...args}
    />
  );
};
