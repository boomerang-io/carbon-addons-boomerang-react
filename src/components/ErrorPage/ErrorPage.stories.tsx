import React from "react";
import GraphicWrangler from "../GraphicWrangler";
import ErrorPage from "./ErrorPage";

export default {
  title: "Errors/ErrorPage",
  component: ErrorPage,
  parameters: {
    docs: {
      description: {
        component:
          "Customizable full page error for Boomerang theme. Used as base component for core 403 and 404 error components",
      },
    },
  },
};

export const Default = (args: any) => {
  return <ErrorPage header={"Header"} title={"Title"} message={"Message"} graphic={<GraphicWrangler />} {...args} />;
};

export const MessageLink = (args: any) => {
  return (
    <ErrorPage
      title={"Title"}
      message={
        <p>
          Hello there, <a href="https://useboomerang.io">use Boomerang!</a>
        </p>
      }
      graphic={<GraphicWrangler />}
      {...args}
    />
  );
};
