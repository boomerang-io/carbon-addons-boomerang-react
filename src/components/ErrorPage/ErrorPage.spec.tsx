import React from "react";
import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import ErrorPage from "./ErrorPage";

const header = "Error";
const title = "Something is wrong";
const message = "Sorry!";

describe("ErrorPage", () => {
  test("snapshot", async () => {
    const { baseElement } = render(<ErrorPage header={header} title={title} message={message} />);
    expect(baseElement).toMatchSnapshot();
  });

  test("functional", async () => {
    const { getByText } = render(<ErrorPage header={header} title={title} message={message} />);
    expect(getByText(header)).toBeInTheDocument();
    expect(getByText(title)).toBeInTheDocument();
    expect(getByText(message)).toBeInTheDocument();
  });

  test("a11y", async () => {
    const { container } = render(<ErrorPage header={header} title={title} message={message} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
