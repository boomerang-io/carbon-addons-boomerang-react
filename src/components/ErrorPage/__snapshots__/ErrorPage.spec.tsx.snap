// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorPage > snapshot 1`] = `
<body>
  <div>
    <div
      class="cds--bmrg-error-page"
    >
      <header
        class="cds--bmrg-error-page__header"
      >
        Error
      </header>
      <h1
        class="cds--bmrg-error-page__title"
      >
        Something is wrong
      </h1>
      <section
        class="cds--bmrg-error-page__message"
      >
        Sorry!
      </section>
    </div>
  </div>
</body>
`;
